packages/calendar/src/Calendar.ts

```diff
@@ -233,12 +191,22 @@ export class Calendar extends SpectrumElement {
         if (changesDates) {
             this.convertToCalendarDates();
             this.checkDatePropsCompliance(changesMin || changesMax);
+            if (this.value) this.currentDate = this.value as CalendarDate;
+            else {
+                const isTodayNonCompliant = this.isNonCompliantDate(this.today);
+
+                if (isTodayNonCompliant) {
+                    if (this.min) this.currentDate = this.min as CalendarDate;
+                    else if (this.max)
+                        this.currentDate = this.max as CalendarDate;
+                } else this.currentDate = this.today;
+            }
```

Comment:
Consider moving these nested ifs in a single method to improve readability. 
Something like 
```

    setDate() {
        if (this.value){
            this.currentDate = this.value as CalendarDate;
            return;
        } 

        if(this.isNonCompliantDate(this.today) && this.min) {
            this.currentDate = this.min as CalendarDate;
            return;
        }

        if(this.isNonCompliantDate(this.today) && this.max) {
            this.currentDate = this.max as CalendarDate;
            return;
        }

        this.currentDate = this.today;
    }
```